lib/new-syntax/keywords.ts

```diff
@@ -49,6 +49,7 @@ export const KEYWORDS = new Set<string>([
     'opt',
     'req',
     'sort',
+    'on',
```

Comment:
'on' is not a keyword there is no special join filter syntax, right?

(maybe we should keep it as reserved word for the future anyway though)

Reply:
yeah, that's what I thought, it's probably good to keep it as keyword